### PR TITLE
Feature/link command

### PR DESCRIPTION
**Pull request by igord**

### CODE REVIEW

**Comment by operatino** [comment]
Most likely we will add other command which should be also called `bb link`, but will follow `bower link` workflow + additional steps (rjs conf generation, model sync etc). How do you think, which command is better to rename, yours or future one that will be wrapping `bower link`?

I propose renaming your command to `bb ln` (as `ln -s`). Combining both linking command will be a bit overhead.

File: lib/portalUtils.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ _.extend(exports, {
     },
     getPom: function(c, p) {
         p = p || '';
-        var pomPath = path.join(c.config._local.path, p, '/pom.xml');
+        var pomPath = path.join(c.config.bbrc.path, p, '/pom.xml');
```
Comment (operatino) [inline lib/portalUtils.js]: Let's remove this util, since it's not used. You can move it to some branch.

File: README.md [hunk 2]
```diff
@@ -186,6 +186,46 @@ Where `.bbrc` file contains this conf:
...
+
+If `--lp-portal` path is set, target will be:
+
+`{cxp path}/src/main/webapp/static/launchpad/{bundle}/widgets/{package name}`
+
+As LP convention, if package name starts with `widget-` it will be stripped out when creating a symlink.
+
+If `--cxp` path is set, target will be:
+
+`{cxp path}/src/main/webapp/static/widgets/{package name}`
```
Comment (operatino) [inline README.md]: For me its not that clear that `cxp path` is the path to portal. CXP is not only the portal, but contentservices, orchestrator all together.

File: README.md [hunk 3]
```diff
@@ -186,6 +186,46 @@ Where `.bbrc` file contains this conf:
...
+
+```
+bb link --source /repo/path -- target /path/to/portalserver/static/dir/
+```
+
+This command also supports conventions used in Launchpad and ES. For example:
+
+If `--lp-trunk` path is set, target will be:
+
+`{lp path}/launchpad-bundles/static/launchpad/{bundle}/widgets/{package name}`
```
Comment (operatino) [inline README.md]: So where from `launchpad-bundles` is coming from?

Comment (igord) [inline README.md]: That is the structure of the launchpad trunk repo. It doesn't follow convention that we are used to.

File: README.md (hunk 2 above)
Comment (igord) [inline README.md]: Yes, it could be just --portal.

**Comment by igord** [comment]
Regarding bower/npm link, I doubt that we will use them at the end. Those are just wrappers around ln command and specific to bower/npm as they always create link inside bower_components/node_modules. That could be changed by editing rc files but it is additional step and it will not work well if you have nested structure as current structure of LP is.

**Comment by operatino** [comment]
Bower link is not just a wrapper, it also resolves the dependencies and does the merging. So we will need it definitely. Ben agrees about it and is voting for renaming this command to `bb ln`.

**Pull request merged**